main.c

```diff
@@ -667,9 +762,49 @@ static void parse_args(int argc, char *const argv[], struct options *options)
 #undef OUTBOUND_ADDR6
 #undef DISABLE_DNS
 #undef MACADDRESS
+#undef TARGET_TYPE
+
+    /* BESS mode */
+    if (options->target_type != NULL &&
+        strcmp(options->target_type, "bess") == 0) {
+        if (argc - optind < 1) {
+            goto error;
+        }
+        if (argc - optind > 1) {
+            fprintf(stderr, "too many arguments\n");
+            goto error;
+        }
+        options->bess_socket = strdup(argv[optind]);
+        if (options->do_config_network) {
+            fprintf(stderr, "--configure conflicts with --target-type=\n");
+            goto error;
+        }
+        if (options->netns_type != NULL) {
+            fprintf(stderr, "--netns-type conflicts with --target-type=bess\n");
+            goto error;
+        }
+        if (options->userns_path != NULL) {
+            fprintf(stderr,
+                    "--userns-path conflicts with --target-type=bess\n");
+            goto error;
+        }
+        printf("WARNING: BESS mode is experimental\n");
+        return;
+    }
+
+    /* NetNS mode*/
+    if (options->target_type != NULL &&
+        strcmp(options->target_type, "netns") != 0) {
+        fprintf(stderr, "--target-type must be either \"netns\" or \"bess\"\n");
+        goto error;
+    }
     if (argc - optind < 2) {
         goto error;
     }
+    if (argc - optind > 2) {
+        // not an error, for preventing potential compatibility issue
+        printf("WARNING: too many arguments\n");
```

Comment:
should this go to stderr?

Reply:
We have been using stdout or printing WARNING messages (why? 😅 )
https://github.com/rootless-containers/slirp4netns/blob/f2ae176d17db4ac044a8431e90cf8c9de5144584/main.c#L563-L604